### ISSUE TITLE
Observed inconsistency in remote-exec

### ISSUE THREAD

**Issue by Gary-Armstrong** [hashicorp/terraform#5806]
I have a set of five "things" that I want done to instances on instantiation.  I'm pretty sure my TF code is solid:
```
  provisioner "remote-exec" {
    connection {
      user = "gary.armstrong"
      host = "${element(aws_instance.viewer.*.private_ip, count.index)}"
    }
    inline = [
      "sudo mkdir -p /etc/chef/ohai/hints && echo 'Created directory /etc/chef/ohai/hints' || echo 'Failed to create /etc/chef/ohai/hints'",
      "sudo touch /etc/chef/ohai/hints/ec2.json && echo 'Touched /etc/chef/ohai/hints/ec2.json' || echo 'Failed to touch /etc/chef/ohai/hints/ec2.json'",
      "sudo rm -f /etc/chef/client.pem && echo 'Removed /etc/chef/client.pem' || echo 'Failed to remove /etc/chef/client.pem'",
      "sudo touch /etc/chef/reboot && echo 'Touched /etc/chef/reboot' || echo 'Failed to touch /etc/chef/reboot'",
      "sudo hostname ${var.project}-${var.environment}-viewer-${format("%03d", count.index)} && echo 'Set hostname' || echo 'Failed to set hostname'"
    ]
  }
```
I definitely have the passwordless sudo set up.  I also know that ssh and sudo and such work because I've seen evidence of some of those commands executing on the instance.  They don't ALWAYS execute, however.

I've been testing the above debuggy TF code today and I'm consistently seeing:
```
null_resource.remote-exec_provision_viewer (remote-exec): Connected!
null_resource.remote-exec_provision_viewer (remote-exec): Created directory /etc/chef/ohai/hints
null_resource.remote-exec_provision_viewer: Creation complete
```
Expected to see a bit more.

Reviewing trace output and other data, all I can really tell is:
1. The inline[] data is put into a /tmp/*.sh file and then executed.
2. STDOUT comes from the first command only.
3. The chef provisioner manages to send all STDOUT(?).

Next step for me is to try other veil-piercing methods, like redirecting STD* to files, but I feel like I should post something on here first and see if I get any bites.

**Comment by phinze** [comment]
Hi @Gary-Armstrong - this looks like the shape of a bug that snuck into 0.6.13 and was fixed in 0.6.14 - what version of Terraform are you using when you see this behavior?

**Comment by Gary-Armstrong** [comment]
Using 0.6.13 but give me a few minutes and I'll be using 0.6.14!

**Comment by Gary-Armstrong** [comment]
This is fixed in 0.6.14
